lib/internal/http2/compat.js

```diff
@@ -654,29 +651,25 @@ class Http2ServerResponse extends Stream {
   }
 
   write(chunk, encoding, cb) {
-    const state = this[kState];
-
     if (typeof encoding === 'function') {
       cb = encoding;
       encoding = 'utf8';
     }
 
-    if (typeof cb !== 'function')
-      cb = nop;
-
-    let err;
-    if (state.ending) {
-      err = new ERR_STREAM_WRITE_AFTER_END();
-    } else if (state.destroyed) {
-      err = new ERR_STREAM_DESTROYED('write');
-    } else if (state.closed) {
-      err = new ERR_HTTP2_INVALID_STREAM();
-    }
-
-    if (err) {
-      process.nextTick(cb, err);
+    if (this[kState].ending) {
+      const err = new ERR_STREAM_WRITE_AFTER_END();
+      if (typeof cb === 'function')
+        process.nextTick(cb, err);
       this.destroy(err);
-      return;
+      return false;
+    } else if (this[kState].closed) {
```

Comment:
My other outstanding question regarding `closed` stands. Might be good to get @jasnell @addaleax to weigh in too. Seems like this existing error is not in the spirit of compatibility which could cause people issues...

(Could also be left for a separate PR.)

Reply:
(It was mentioned in the linked issue, hence me bringing it up.)

Reply:
I'll leave this as is until there is more input.

Reply:
I think the biggest issue was the `throw`.